src/bin/scheduler/setup/general.rs

```diff
@@ -1,50 +1,160 @@
     setup_results_directories(global_config, &plans)?;
     setup_managed_directories(&global_config.managed_directory, &plans)?;
 
-    let mut surviving_plans: Vec<Plan>;
-    let mut general_setup_failures = GeneralSetupFailures::default();
-    (
-        surviving_plans,
-        general_setup_failures.working_directory_permissions,
-    ) = adjust_working_directory_permissions(global_config, plans);
+    let mut surviving_plans: Vec<Plan> = setup_working_directories(global_config, plans)?;
+    sort_plans_by_grouping(&mut surviving_plans);
+    Ok(surviving_plans)
+}
 
-    general_setup_failures.write(
+fn setup_working_directories(
+    global_config: &GlobalConfig,
+    plans: Vec<Plan>,
+) -> AnyhowResult<Vec<Plan>> {
+    let (surviving_plans, plan_failures) = setup_plans_working_directory(plans);
+    let (surviving_plans, rcc_failures) =
+        setup_rcc_working_directories(&global_config.working_directory, surviving_plans);
+    GeneralSetupFailures {
+        working_directory_permissions: plan_failures
+            .into_iter()
+            .chain(rcc_failures.into_iter())
+            .collect(),
+    }
+    .write(
         global_config
             .results_directory
             .join("general_setup_failures.json"),
         &global_config.results_directory_locker,
     )?;
-
-    sort_plans_by_grouping(&mut surviving_plans);
     Ok(surviving_plans)
 }
 
-fn setup_working_directories(working_directory: &Utf8Path, plans: &[Plan]) -> AnyhowResult<()> {
-    if working_directory.exists() {
-        remove_dir_all(working_directory).context("Failed to remove working directory")?;
+fn setup_plans_working_directory(plans: Vec<Plan>) -> (Vec<Plan>, HashMap<String, String>) {
+    let mut surviving_plans = Vec::new();
+    let mut failures = HashMap::new();
+    for plan in plans.into_iter() {
+        if let Err(e) = create_dir_all(&plan.working_directory) {
+            let error = anyhow!(e).context(format!(
+                "Failed to create working directory {} of plan {}",
+                plan.working_directory, plan.id
+            ));
+            info!("{error:#}");
+            failures.insert(plan.id.clone(), format!("{error:#}"));
+            continue;
+        }
+        if let Session::User(user_session) = &plan.session {
+            info!(
+                "Granting full access for {} to user `{}`.",
+                &plan.working_directory, &user_session.user_name
+            );
+            if let Err(e) = grant_full_access(&user_session.user_name, &plan.working_directory) {
+                let error = anyhow!(e).context(format!(
+                    "Failed to set permissions for working directory {} of plan {}",
+                    plan.working_directory, plan.id
+                ));
+                info!("{error:#}");
+                failures.insert(plan.id.clone(), format!("{error:#}"));
+                continue;
+            };
+        }
+        surviving_plans.push(plan);
```

Comment:
We should also put this into a separate function, since it's repeated in `setup_with_one_directory_per_user` if I am not mistaken

Reply:
I don't think so? `setup_with_one_directory_per_user` creates one directory per user, and is only used during the setup phase.
`setup_plans_working_directory` sets up one directory per plan.

Reply:
True, sry, my bad